src/cloudai/systems/slurm/slurm_system.py

```diff
@@ -142,6 +184,49 @@ class SlurmSystem(BaseModel, System):
 
     data_repository: Optional[DataRepositoryConfig] = None
 
+    def model_post_init(self, __context: Any) -> None:
+        del __context
+        try:
+            best = self.best_account
+            if self.account is None:
+                if best:
+                    self.account = best
+                    logging.info(f"Auto-selected Slurm account: {best}")
+            elif best and self.account != best:
+                logging.warning(f"Selected account '{self.account}' is suboptimal; '{best}' is better.")
+        except Exception:
+            pass
+
+    @property
+    def best_account(self) -> Optional[str]:
+        stdout, _ = self.fetch_command_output("sshare --parsable")
+        entries = self.parse_sshare_output(stdout)
+        if not entries:
+            return None
+        return max(entries, key=lambda e: e.fairshare).account
+
+    def parse_sshare_output(self, output: str) -> List[ShareEntry]:
```

Comment:
I suggest we move this one into `ShareEntry`, interface can be like `ShareEntry().best_account`. `SlurmSystem` shouldn't care about constructing it correctly, maybe in future we will rely on some REST/protobuf based API with `sshare`.